extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetKeyValuePartitionedLayout.java

```diff
@@ -88,8 +89,16 @@ public ParquetKeyValuePartitionedLayout(
     public final void findKeys(@NotNull final Consumer<ParquetTableLocationKey> locationKeyObserver) {
         final SeekableChannelsProvider provider = SeekableChannelsProviderLoader.getInstance().fromServiceLoader(
                 tableRootDirectory, readInstructions.getSpecialInstructions());
+        final boolean isFileURI = FILE_URI_SCHEME.equals(tableRootDirectory.getScheme());
         try (final Stream<URI> uriStream = provider.walk(tableRootDirectory)) {
-            final Stream<URI> filteredStream = uriStream.filter(ParquetUtils::isVisibleParquetURI);
+            final Stream<URI> filteredStream = uriStream.filter(uri -> {
+                if (isFileURI) {
+                    final URI relativeUri = tableRootDirectory.relativize(uri);
+                    return isVisibleParquetFile(new File(relativeUri.getPath()));
```

Comment:
So, I think this code is technically correct, but it's potentially dangerous b/c this file is relative to the _working_ directory (as opposed to relative to tableRootDirectory). It's technically correct b/c we are only using the file as a placeholder for traversing the parent / names.

I think a more explicit method that takes this into account would be better: `isVisibleParquetFile(Path rootDir, Path file)`, with the iteration stopping once rootDir (as opposed to null) is hit. I also am generally in favor of using Path in these situations as it's a more general API and hints a bit more that we don't need to do any stat-ing to return a result.